wordpress-mobile/WordPress-Android
# Populate Scan History screen with data from /scan/history endpoint

Parent issue #13326

Related PR in FluxC - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1841

Merge instructions
1. Use FluxC tag
2. Remove "Not ready for merge" label
3. Merge this PR

This PR uses data provided by the /scan/history endpoint to populate Scan History screen (note: filtering by tab is not yet supported).
We decided to fetch data in the ScanHistoryViewModel (parent) and provide them to ScanHistoryListViewModel (children) through LiveData. Filtering is not yet supported, but I plan to do the actually filtering in the child view models.

To test:
1. Make sure Scan feature flag is enabled
2. Open Scan screen
3. Click on "History" button in the Toolbar
4. Notice the history screen is shown and threats get displayed after the request finished (it sometime takes 20+ seconds)

Note: 
- loading/error states are not implemented
- tab filtering doesn't work yet
- on item click is not implemented

PR submission checklist:

- [] I have considered adding unit tests where possible.
